libs/lua-hash/src/hash.c

```diff
@@ -0,0 +1,47 @@
+#include <libubox/md5.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MD5_DIGEST_LEN 16
+#define MD5_STRLEN 2*MD5_DIGEST_LEN+1
+
+static int md5(lua_State *L) {
+	char *input = (char *) lua_tostring(L, 1);
```

Comment:
This should use luaL_checklstring (so only strings are accepted); the "lstring" version has an additional argument for the length, so strings with embedded zeros can be supported.

The cast chould be removed, *input* should be a `const char *`.